.gitignore

Copyright (C) 2013 Curt Brune <curt@cumulusnetworks.com>
Copyright (C) 2017 david_yang <david_yang@accton.com>

SPDX-License-Identifier:     GPL-2.0

diff --git a/.gitignore b/.gitignore
new file mode 100644
index 000000000000..8e70baa4dc12
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,83 @@
+#
+# Kbuild ignores
+#
+.*
+*.o
+*.o.*
+*.a
+*.s
+Kbuild
+Config.in
+
+#
+# Never ignore these
+#
+!.gitignore
+
+#
+# Normal output
+#
+/busybox
+/busybox_old
+/busybox_unstripped*
+
+#
+# Backups / patches
+#
+*~
+*.orig
+*.rej
+/*.patch
+
+#
+# debugging stuff
+#
+core
+.gdb_history
+.gdbinit
+
+#
+# testing output
+#
+/busybox.links
+/runtest-tempdir-links
+/testsuite/echo-ne
+
+#
+# cscope output
+#
+cscope.files
+cscope.in.out
+cscope.out
+cscope.po.out
+
+# the following files are generated automatically
+# after ONIE built
+/applets/applet_tables
+/applets/usage
+/applets/usage_pod
+/docs/BusyBox.html
+/docs/BusyBox.txt
+/docs/busybox.1
+/docs/busybox.net/
+/docs/busybox.pod
+/include/NUM_APPLETS.h
+/include/applet_tables.h
+/include/applets.h
+/include/autoconf.h
+/include/bbconfigopts.h
+/include/bbconfigopts_bz2.h
+/include/common_bufsiz.h
+/include/common_bufsiz.h.method
+/include/config/
+/include/usage.h
+/include/usage_compressed.h
+/scripts/basic/docproc
+/scripts/basic/fixdep
+/scripts/basic/split-include
+/scripts/kconfig/conf
+/scripts/kconfig/lex.zconf.c
+/scripts/kconfig/lxdialog/lxdialog
+/scripts/kconfig/mconf
+/scripts/kconfig/zconf.hash.c
+/scripts/kconfig/zconf.tab.c
